protocols/api/src/main/java/org/apache/james/protocols/api/ProtocolSession.java

```diff
@@ -148,10 +148,39 @@ public String toString() {
 
     
     /**
-     * Return the {@link InetSocketAddress} of the remote peer
+     * Return the {@link InetSocketAddress} of the remote peer. If proxy support
+     * is enabled, then it returns the remote peer given by the proxy.
      */
     InetSocketAddress getRemoteAddress();
 
+    /**
+     * Return true if PROXY is required by the configuration
+     *
+     * @return supported
+     */
+    boolean isProxyRequired();
```

Comment:
Doing a search on `isProxyRequired()` in this PR I do not see where this method is called, except in tests. Can we get rid of it?

Reply:
Seems legit.